### PR TITLE
feat: return faulty sectors

### PR DESCRIPTION
**Pull request by vmx**
If Window PoSt fails, return the faulty sectors. They are part of the
error response. The `faulty_sectors_len` is the number of faulty sectors
(0 if there aren't any faulty sectors) and `faulty_sectors_ptr` points
to an array of unsigned 64-bit integers, representing the sector IDs.

### CODE REVIEW

File: rust/src/proofs/api.rs [hunk 1]
```diff
@@ -1722,4 +1735,169 @@ pub mod tests {
...
+                randomness,
+                private_replicas.as_ptr(),
+                private_replicas.len(),
+                prover_id,
+            );
+
+            assert_eq!(
+                (*resp_j).status_code,
+                FCPResponseStatus::FCPUnclassifiedError,
+                "generate_winning_post should have failed"
```
Comment (porcuquine) [inline rust/src/proofs/api.rs:1881]: 'generate_window_post', I think.

**Review by porcuquine — APPROVED** [review]

**Comment by vmx** [comment]
This should *NOT* be mereged until the corresponding Go changes (which should be pretty minimal) are in.

**Review by magik6k — APPROVED** [review]

File: proofs.go [hunk 2]
```diff
@@ -808,6 +808,14 @@ func fromFilPoStFaultySectors(ptr []uint64, l uint) ([]abi.SectorNumber, error)
 		return nil, nil
 	}
 
+	type sliceHeader struct {
+		Data unsafe.Pointer
+		Len  int
+		Cap  int
+	}
+
+	(*sliceHeader)(unsafe.Pointer(&ptr)).Len = int(l) // don't worry about it
```
Comment (Kubuxu) [inline proofs.go:817]: Is this needed? Isn't just the fact that we are inerating up to `l` below enough?

Comment (magik6k) [reply proofs.go:817]: The generated cgo code doesn't set it, si we panic on out-of-bounds

**Review by whyrusleeping — APPROVED** [review]

**Pull request merged**